pyproject.toml

```diff
@@ -0,0 +1,87 @@
+[tool.isort]
+known_first_party = [
+    "lightgbm"
+]
+line_length = 120
+skip_glob = [
+    "*/external_libs/*",
+    "*/lightgbm-python/*"
+]
+
+[tool.ruff]
+exclude = [
+    "build",
+    "compile",
+    "external_libs",
+    "lightgbm-python",
+]
+line-length = 120
+
+# this should be set to the oldest version of python LightGBM supports
+target-version = "py37"
+
+[tool.ruff.format]
+docstring-code-format = false
+exclude = [
+    "build/*.py",
+    "compile/*.py",
+    "examples/*.py",
+    "external_libs/*.py",
+    "lightgbm-python/*.py",
+    "python-package/*.py",
+    "tests/*.py"
+]
+indent-style = "space"
+quote-style = "double"
+
+[tool.ruff.lint]
```

Comment:
The introduction of this table (and some other renamings) are in response to this warning from `ruff`:

```text
warning: The top-level linter settings are deprecated in favour of their counterparts in the `lint` section. Please update the following options in `pyproject.toml`:
  - 'ignore' -> 'lint.ignore'
  - 'select' -> 'lint.select'
  - 'pydocstyle' -> 'lint.pydocstyle'
  - 'per-file-ignores' -> 'lint.per-file-ignores'
```

Which have been showing up in the `lint` CI job, e.g. https://github.com/microsoft/LightGBM/actions/runs/7852675526/job/21431398282#step:3:405